src/init.cpp

```diff
@@ -958,6 +958,11 @@ void InitParameterInteraction()
             LogPrintf("%s: parameter interaction: -externalip set -> setting -discover=0\n", __func__);
     }
 
+    if (mapArgs.count("-recoverwallet")) {
+        if (SoftSetBoolArg("-rescan", true))
+            LogPrintf("%s: parameter interaction: -recoverwallet=1 -> setting -rescan=1\n", __func__);
+    }
+
     if (GetBoolArg("-salvagewallet", false)) {
```

Comment:
`-recoverwallet` should be mutually exclusive with `-salvagewallet` and probably with `-zapwallettxes`.